src/components/Topics/Topics.jsx

```diff
@@ -0,0 +1,64 @@
+import './Topics.css'
+import { useParams} from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import newsApi from '../utils/newsAPI';
+import { Link } from 'react-router-dom';
+
+function formatDate(createdAtDate) {
+    const date = new Date(createdAtDate);
+    return date.toLocaleDateString('en-GB');
+}
+
+
+const Topics = () => {
+    const params = useParams();
+    const topic = params.topic
+
+    const [allArticles, setAllArticles] = useState([])
+    
+    useEffect(() => {
+        newsApi.get('/articles')
+        .then((response) => {
+            const data = response.data.article
+            setAllArticles(data)
+            
+        })
+    }, [])
+
+    const topicArticles = []
+
+    {allArticles.map((article) => {
```

Comment:
rather than filtering the topics, we want to be able to fetch the articles by topic so we have the most up to date list of articles for each topic selected. Consider saving the selected topic to a state and then making a new api request based on the topic selected and updating the articles with the results that come back from this